examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/OfferingFragment.kt

```diff
@@ -130,6 +198,76 @@ class OfferingFragment : Fragment(), PackageCardAdapter.PackageCardAdapterListen
         }
     }
 
+    override fun onAddOnPurchaseClicked(selectedPackages: List<Package>) {
+        if (Purchases.sharedInstance.finishTransactions) {
+            startAddOnPurchase(selectedPackages)
+        } else {
+            startAddOnPurchaseWithoutFinishingTransaction(selectedPackages)
+        }
+    }
+
+    override fun onSelectionChanged(hasSelectedPackages: Boolean, hasValidBaseProduct: Boolean) {
+        updatePurchaseButtonState(hasSelectedPackages, hasValidBaseProduct)
+    }
+
+    private fun updatePurchaseButtonState(hasSelectedPackages: Boolean, hasValidBaseProduct: Boolean) {
+        val isEnabled = binding.isAddOnPurchaseMode == true && hasSelectedPackages && hasValidBaseProduct
+        binding.isPurchaseButtonEnabled = isEnabled
+    }
+
+    @OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+    private fun startAddOnPurchase(selectedPackages: List<Package>) {
+        toggleLoadingIndicator(true)
+        val basePackage = packageCardAdapter?.getBaseProduct() ?: selectedPackages.first()
+        val addOnPackages = selectedPackages.filter { it != basePackage }
+
+        val purchaseParamsBuilder = PurchaseParams.Builder(
+            activity = requireActivity(),
+            packageToPurchase = basePackage,
+        )
+            .addOnPackages(addOnPackages = addOnPackages)
+
+        selectedReplacementMode?.let {
+            purchaseParamsBuilder.googleReplacementMode(it)
+        }
+
+        val purchaseParams = purchaseParamsBuilder.build()
+
+        Purchases.sharedInstance.purchase(
+            purchaseParams = purchaseParams,
+            callback = object : PurchaseCallback {
+                override fun onCompleted(storeTransaction: StoreTransaction, customerInfo: CustomerInfo) {
+                    toggleLoadingIndicator(false)
+                    Toast.makeText(
+                        requireContext(),
+                        "Add-On purchase completed successfully!",
+                        Toast.LENGTH_LONG,
+                    ).show()
+                    findNavController().navigateUp()
+                }
+
+                override fun onError(error: PurchasesError, userCancelled: Boolean) {
+                    toggleLoadingIndicator(false)
+                    if (!userCancelled) {
+                        Toast.makeText(
+                            requireContext(),
+                            "Add-On purchase failed: ${error.message}",
+                            Toast.LENGTH_LONG,
+                        ).show()
+                    }
+                }
+            },
+        )
+    }
+
+    private fun startAddOnPurchaseWithoutFinishingTransaction(selectedPackages: List<Package>) {
+        // For non-finishing transactions, we'll handle each package individually
+        // This is a simplified approach - in a real implementation you might want to handle this differently
+        selectedPackages.forEach { pkg ->
```

Comment:
I think this is wrong...I'll update this to make an add-on purchase instead of purchasing them individually